robottelo/report_portal/portal.py

```diff
@@ -0,0 +1,257 @@
+        :returns dict: The launches of Report portal.
+            if sat_version is given,
+            ```{'snap_version1':launch_object1, 'snap_version2':launch_object2}```
+            else,
+            ```{'sat_version1':{'snap_version1':launch_object1, ..}, 'sat_version2':{}}```
+        """
+        if launch_type not in launch_types:
+            raise ValueError(
+                f'Wrong Launch Type is given as {launch_type} should be one of {launch_types}'
+            )
+        data = self._launch_requester()
+        sorted_launches = sorted(
+            data['content'], key=lambda launch: launch['start_time'], reverse=True
+        )
+        typed_launches = filter(
+            lambda launch: launch['name'].lower() == launch_type, sorted_launches
+        )
+        launches_ = self._format_launches(typed_launches)
+        if sat_version:
+            if sat_version not in launches_:
+                raise ValueError(
+                    f'The given satellite version \'{sat_version}\' is not available in '
+                    f'Report portal project \'{self.rp_project}\''
+                )
+            else:
+                return launches_[sat_version]
+        return launches_
+
+    def launch(self, sat_version, snap_version=None, launch_type='satellite6'):
+        """Returns a specific launch data in Report Portal Project
+
+        This does not includes each tests data in launch
+
+        :param str sat_version: The satellite version
+        :param str snap_version: The snap version of a given satellite version
+            if None, the latest launch data of a given sat_version is returned
+        :param str launch_type: Either satellite6 or upgrades, default returns only
+            non-upgrade launches
+        :returns dict: The data directory of requested or latest launch
+        """
+        launches_ = self.launches(sat_version=sat_version, launch_type=launch_type)
+        if snap_version:
+            if snap_version not in launches_.keys():
+                raise ValueError(
+                    f'The given snap_version \'{snap_version}\' is not available '
+                    f'in satellite version \'{sat_version}\''
+                )
+        else:
+            # Get the latest launch of the given sat_version
+            snap_version = next(iter(launches_.keys()))
+        return launches_[snap_version]
+
+
+class Launch:
+    def __init__(self, rp, launch_info):
+        """Initiates the Launch and its properties of a specified snap version in
+        a specified satellite version
+
+        :param ReportPortal rp: A report portal object
+        :param dict launch_info: The dict information about a launch
+        """
+        self.report_portal = rp
+        self.info = launch_info
+        self.name = self.info['name']
+        self.statistics = self.info['statistics']['executions']
+        self._versions(self.info)
+
+    def _versions(self, launch_info):
```

Comment:
Not a blocker - but `launch_info` doesn't have to be passed, it can be accessed with `self.info`

Reply:
Make Sense, wonder how I missed :)